### PR TITLE
Fix 3rd party signin

### PR DESCRIPTION
**Pull request by vanwagonet**
refs: MBL-12493
affects: Parent, Student, Teacher
release note: Fixed a login bug with 3rd party providers

### CODE REVIEW

File: Core/Core/Login/LoginWebViewController.swift [hunk 1]
```diff
@@ -18,7 +18,11 @@ import UIKit
 import WebKit
 
 class LoginWebViewController: UIViewController, LoginWebViewProtocol {
-    let webView = WKWebView(frame: UIScreen.main.bounds)
+    lazy var webView: WKWebView = {
+        let configuration = WKWebViewConfiguration()
+        configuration.websiteDataStore = .nonPersistent()
+        return WKWebView(frame: UIScreen.main.bounds, configuration: configuration)
+    }()
```
Comment (vanwagonet) [inline Core/Core/Login/LoginWebViewController.swift]: Allows re-login with new credentials.

File: Core/Core/Login/LoginWebPresenter.swift [hunk 2]
```diff
@@ -109,9 +109,10 @@ class LoginWebPresenter {
             return .cancel
         }
 
-        // I dunno why, but we have to wait for the code to be the first param cuz it can keep changing as we follow redirects
-        //  "/canvas/login?code="
-        if let code = codeFromQueryItems(queryItems), let mobileVerify = mobileVerifyModel, let baseURL = mobileVerify.base_url {
+        if // wait for "https://canvas/login?code="
+            url.absoluteString.hasPrefix("https://canvas/login"),
+            let code = queryItems?.first(where: { $0.name == "code" })?.value, !code.isEmpty,
+            let mobileVerify = mobileVerifyModel, let baseURL = mobileVerify.base_url {
```
Comment (vanwagonet) [inline Core/Core/Login/LoginWebPresenter.swift]: A more robust check than "is code the first query item?"

this is a weird url though, so `components.host` and `components.path` are both nil, which is why we check `absoluteString`.

**Comment by inst-danger** [comment]
#### Release Note: 
Fixed a login bug with 3rd party providers

#### Affected Apps: Parent, Student, Teacher

[MBL-12493](https://instructure.atlassian.net/browse/MBL-12493)

Coverage | New % | Master % | Delta
-------- | ----- | -------- | -----
**Core** | 95.15% | 95.14% | 0.01%
**Student** | 92.86% | 92.86% | 0%
**Teacher** | 100% | 100% | 0%
**React Native** | 92.67% | 92.67% | 0%
<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by shichongrui** [comment]
Looks good, testing.

**Pull request merged**